server/main.lua

```diff
@@ -59,36 +59,64 @@ end)
 ----   Functions   ----
 -----------------------
 
+--- Gives the user the keys to the vehicle
+--- @param source number ID of the player
+--- @param plate string The plate number of the vehicle.
 function GiveKeys(source, plate)
+    local citizenid = getCitizenId(source)
+
+    if not citizenid then return end
+
     local keys = Player(source).state.keysList or {}
 
     if keys[plate] then return end
     keys[plate] = true
-    Player(source).state:set('keysList', keys, true)
-    local citizenid = getCitizenId(source)
 
-    if not citizenid then return end
+    Player(source).state:set('keysList', keys, true)
 
     if not keysList[citizenid] then
-        keysList[citizenid] = {}
+        keysList[citizenid] = {plate = true}
+    else
+        keysList[citizenid][plate] = true
     end
 
-    keysList[citizenid][plate] = true
-
     exports.qbx_core:Notify(source, locale('notify.keys_taken'))
 end
 
+RegisterNetEvent('qbx_vehiclekeys:server:giveKeys', function(source, plate)
+    if not GetInvokingResource() then return end
+    GiveKeys(source, plate)
+end)
+
 exports('GiveKeys', GiveKeys)
 
+--- Removing the vehicle keys from the user
+--- @param source number ID of the player
+--- @param plate string The plate number of the vehicle.
 function RemoveKeys(source, plate)
-    local state = Player(source).state
-
-    if not state.keysList[plate] then return end
     local citizenid = getCitizenId(source)
-    state.keysList = keysList[citizenid]
-    keysList[citizenid][plate] = nil
+
+    if not citizenid then return end
+
+    local keys = Player(source).state.keysList or {}
+
+    if not keys[plate] then return end
```

Comment:
Maybe we should lib.print.warn before returning end, or at the very least lib.print.verbose would be appropriate.

Reply:
console alerts should be added, however, a good time would be after the main work is completed